lib/atscppapi/src/utils_internal.cc

```diff
@@ -99,6 +97,8 @@ handleTransactionEvents(TSCont cont, TSEvent event, void *edata)
 void
 setupTransactionManagement()
 {
+  // Reserve a transaction slot
+  TSAssert(TS_SUCCESS == TSHttpArgIndexReserve("atscppapi", "ATS CPP API", &TRANSACTION_STORAGE_INDEX));
```

Comment:
I think the problem with `GlobalPlugin::GlobalPlugin` is it can be called multiple times, once for each global plugin defined. I think only one transaction arg is required, though, for all global plugins.